### DISCUSSION TITLE
Is there away to redirect a search if a paricular entry or word is entered?

### DISCUSSION THREAD

**Discussion by tedtninety** [microsoft-search/pnp-modern-search#3152] [Q&A] [answered]
When a user inputs "HR' or "hr" , a bevy of results that remotely refer or include "hr" returns.
However, I'd like to forward them to a URL that uses a refinablestring for the category "hr".
I have that working URL, but how do I get it to automatically forward to it when a user inputs "hr"?
Is there a way or  another solution altogether?

Perhaps, JavaScript? If so what, what is the name of the input text field for JS to watch for the entry?

Thanks for any assistance.

**Comment by kasperbolarsen** [comment, marked as answer]
Have you considered using ranking to get the most relevant hit to the top. Like boosting results that contains the RefinableString:HR.

**Reply by tedtninety** [reply]
Hi @kasperbolarsen , I hadn't considered that, I went ahead and read up on ranking , however I am a bit lost with the syntax :

With this example to follow, how do I replace the values with this example?
_(cat OR dog) XRANK(cb=100) thoroughbred_

**(HR or hr) XRANK(cb=100) RefinableString 42** 
Would that be correct?

And lastly ,if I already have items in the Query Template
For example :
({SearchTerms})
path:https://domain.sharepoint.com/sites/contosolandings/sitepages/LR1/*

What is the syntax to  add the above, just a semicolon to separate the two? 

Thank you

**Reply by wobba** [reply]
Put this in the query template field: `{searchTerms} XRANK(stdb=5) RefinableString123:({searchTerms})`

where you use the refinablestring property you have mapped to. This will not auto-redirect on special terms, just boost results. If you want to auto-redirect on a term you need to develop some custom component. I would also advice looking into the old query rules and promoted results, or Bookmarks if using the Microsoft Search Graph provider.

@tedtninety maybe you can explain a bit more about the category, refinablestring mapping and logic you want.

**Reply by tedtninety** [reply]
@wobba  Thank you for your reply.

> maybe you can explain a bit more about the category, refinablestring mapping and logic you want.
To make sure that the users find the relevant results when "HR" is entered, I had to find a way to make it so that they did not have to click on a category filter after a search, to show all HR related info that is filtered from a refinablestring for the column "category".

Initially as a solution, I copied the link generated from the filtered  (refinablestring) category for HR"  and pasted a direct link for users looking for "HR" info on the page.
However, I was told that entering "HR" int o the search box - for those who overlooked the direct link, found the results confusing ;with the results not seemingly applicable since the results with anything "hr" displayed returned without any finite relevance.  
Since I am using SharePoint Search, I instead tried to use JavaScript to read the input, and subsequently the URL for parameter "q" value. If it is equal to "HR" then redirect to the aforementioned filter generated link for refinablestring (Category). Unfortunately, it was not a rewarding effort. 

Despite my attempts, if I am not mistaken, the solution you provided with RANKING will effectively display all the entries that are HR - RefinableString -Category on top and all else that has HR in it will be displayed below it, is that correct?

Thank you.

**Reply by wobba** [reply]
It will boost results high up with categories matching all the user term(s). You can play with the number in the boost. And use stdb and not cb. 5 should give a pretty big boost.

**Reply by tedtninety** [reply]
Thank you @wobba and @kasperbolarsen !
